### PR TITLE
Restore ability to create un-focusable buttons

### PR DESCRIPTION
**Pull request by nakambo**
### Platforms Impacted
- [ ] iOS
- [x] macOS
- [x] win32 (Office)
- [x] windows
- [ ] android

### Description of changes

Previously (with the old deprecated buttons) it was possible to create buttons that aren't disabled but still are unfocusable. (this may sound like a bizarre use case, but is handy for creating things like split buttons where the individual component buttons are not focusable).

We seem to have lost this ability with the new Button -- which will override focusable even if passed by the component user. Let's not -- if that is specified, we respect it (and allow people to create un-focusable enabled buttons, or focusable disabled buttons as before).

### Verification

Manually tested the `focusable={false}` prop in the Win32 tester app. Existing buttons in the E2E section that don't have this property are unaffected.

### Pull request checklist

This PR has considered (when applicable):
- [ ] Automated Tests
- [ ] Documentation and examples
- [x] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

**Comment by rurikoaraki** [comment]
@nakambo Thanks for catching this and fixing, appreciate it!

File: apps/fluent-tester/src/TestComponents/Button/E2EButtonTest.tsx [hunk 1]
```diff
@@ -60,6 +62,14 @@ export const E2EButtonExperimentalTest: React.FunctionComponent = () => {
         >
           Press &quot;a&quot; or &quot;b&quot; after focusing this button
         </Button>
+        <Button
+          testID={BUTTON_FOCUSABLE_TEST_COMPONENT}
+          onClick={onClick}
+          accessibilityLabel={BUTTON_FOCUSABLE_TEST_COMPONENT_LABEL}
+          focusable={false}
+        >
+          This button isn&apos;t focusable (but isn&apos;t disabled or indeterminate either)
```
Comment (rurikoaraki) [inline apps/fluent-tester/src/TestComponents/Button/E2EButtonTest.tsx:71]: Is the "&apos;" required? I don't think I've seen it before

Comment (rurikoaraki) [reply apps/fluent-tester/src/TestComponents/Button/E2EButtonTest.tsx:71]: I guess it's right up above in a different button so uh, sure

**Review by rurikoaraki — APPROVED** [review]

File: apps/fluent-tester/src/TestComponents/Button/E2EButtonTest.tsx (hunk 1 above)
Comment (nakambo) [reply apps/fluent-tester/src/TestComponents/Button/E2EButtonTest.tsx:71]: you mean `isn't` directly without escaping it? VSCode seemed to complain about it, I didn't actually try building with that

Comment (nakambo) [reply apps/fluent-tester/src/TestComponents/Button/E2EButtonTest.tsx:71]: that's so strange, so VSCode is picking up some eslint rule from somewhere, but `npm run build` does not complain.

image

image

I'll guess I'll leave it escaped?

**Comment by rurikoaraki** [comment]
Filed an issue to improve the other controls, too. Unfortunately not high on the priority list for me, so it might take a little bit to get to it.

File: apps/fluent-tester/src/TestComponents/Button/E2EButtonTest.tsx (hunk 1 above)
Comment (rurikoaraki) [reply apps/fluent-tester/src/TestComponents/Button/E2EButtonTest.tsx:71]: Yeah that's fine, I was just curious.

**Pull request merged**